File: backend/app/models/enums.py
import enum


class ProcessStatus(str, enum.Enum):
    pending = "pending"
    processing = "processing"
    processed = "processed"
    similarity_checked = "similarity_checked"
    completed = "completed"
    failed = "failed"


class NewsCategory(str, enum.Enum):
    financial_cyber = "金融业网络安全事件"
    major_cyber = "重大网络安全事件"
    data_breach = "重大数据泄露事件"
    vulnerability = "重大漏洞风险提示"
    other = "其他"


class BriefingStatus(str, enum.Enum):
    draft = "draft"
    published = "published"


class LlmTaskType(str, enum.Enum):
    default = "default"
    translate = "translate"
    summarize = "summarize"
    classify = "classify"
    similarity = "similarity"
    importance = "importance"
    embedding = "embedding"

File: frontend/src/types/enums.ts
export const NEWS_CATEGORIES = [
  '金融业网络安全事件',
  '重大网络安全事件',
  '重大数据泄露事件',
  '重大漏洞风险提示',
  '其他',
] as const

export type NewsCategory = typeof NEWS_CATEGORIES[number]

export const PROCESS_STATUSES = [
  'pending',
  'processing',
  'processed',
  'similarity_checked',
  'completed',
  'failed',
] as const

export const CATEGORY_COLORS: Record<string, string> = {
  '金融业网络安全事件': '#1890ff',
  '重大网络安全事件': '#f5222d',
  '重大数据泄露事件': '#fa8c16',
  '重大漏洞风险提示': '#faad14',
  '其他': '#8c8c8c',
}
